sdk/src/cameras/itof-camera/mode_info.cpp

```diff
@@ -139,37 +133,37 @@ aditof::Status ModeInfo::setImagerTypeAndModeVersion(int type, int version) {
     switch (g_imagerType) {
     case 1: {
         if (version == 3) {
-            g_modeInfoData.assign(std::begin(g_newMixedModesAdsd3500),
-                                  std::end(g_newMixedModesAdsd3500));
-            LOG(INFO) << "Using new mixed modes table for adsd3500.";
+            g_modeInfoData.assign(std::begin(g_newMixedModesAdsd3500Adsd3100),
+                                  std::end(g_newMixedModesAdsd3500Adsd3100));
+            LOG(INFO) << "Using new mixed modes table for ADSD3500 w/ADSD3100.";
         } else if (version == 2) {
-            g_modeInfoData.assign(std::begin(g_newModesAdsd3500),
-                                  std::end(g_newModesAdsd3500));
-            LOG(INFO) << "Using new modes table for adsd3500.";
+            g_modeInfoData.assign(std::begin(g_newModesAdsd3500Adsd3100),
+                                  std::end(g_newModesAdsd3500Adsd3100));
+            LOG(INFO) << "Using new modes table for ADSD3500 w/ADSD3100.";
 
         } else if (version == 1) {
             g_modeInfoData.assign(std::begin(g_newModesAdsd3100),
                                   std::end(g_newModesAdsd3100));
-            LOG(INFO) << "Using new modes table for adsd3100.";
+            LOG(INFO) << "Using new modes table for ADSD3100.";
         } else {
-            g_modeInfoData.assign(std::begin(g_oldModes), std::end(g_oldModes));
-            LOG(INFO) << "Using old modes table.";
+            LOG(INFO) << "Unknow modes table.";
+            return Status::INVALID_ARGUMENT;
```

Comment:
We need to catch this error in the initialize function from camera_itof.cpp (the setImagerTypeAndModeVersion is being called in 4 or 5 places without validation).